packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -499,47 +533,64 @@ export class FASTSelect extends FormAssociatedSelect {
      */
     public keydownHandler(e: KeyboardEvent): boolean | void {
         super.keydownHandler(e);
+
         const key = e.key || e.key.charCodeAt(0);
+        let preventDefault = false;
 
         switch (key) {
             case keySpace: {
-                e.preventDefault();
-                if (this.collapsible && this.typeAheadExpired) {
+                if (this.multiple || this.listboxMode) {
+                    this.open = true;
+                } else if (this.collapsible && this.typeAheadExpired) {
                     this.open = !this.open;
                 }
+                preventDefault = true;
                 break;
             }
 
             case keyHome:
             case keyEnd: {
-                e.preventDefault();
+                preventDefault = true;
                 break;
             }
 
             case keyEnter: {
-                e.preventDefault();
-                this.open = !this.open;
+                if (this.multiple || this.listboxMode) {
+                    if (!this.open) {
+                        this.open = true;
+                        break;
+                    }
+                    const option = this._options[this.activeIndex];
+                    option.selected = !option.selected;
+                    preventDefault = true;
+                    break;
+                } else {
+                    this.open = this.collapsible && !this.open;
+                }
+                preventDefault = true;
                 break;
             }
-
             case keyEscape: {
                 if (this.collapsible && this.open) {
-                    e.preventDefault();
                     this.open = false;
+                    preventDefault = true;
                 }
                 break;
             }
 
             case keyTab: {
                 if (this.collapsible && this.open) {
-                    e.preventDefault();
                     this.open = false;
+                    preventDefault = true;
                 }
-
                 return true;
             }
         }
 
+        if (preventDefault) {
```

Comment:
This handler is called from a template binding which marks the event with `preventDefault()` automatically unless the function returns `true`.  So instead of calling `preventDefault()` I would suggest replacing the `return !(key === keyArrowDown || key === keyArrowUp);` line with a ` return !preventDefault`

Reply:
(Personally, I also like starting event handlers with a "if (e.defaulltPrevented) {return}" check if there is any chance an author could have added some custom event handling.)

Reply:
Thanks!